File: IntroToLINQ/IntroToLINQ_CSharp/Program - With EF.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace IntroToLINQ
{
    class Program
    {
        static void Main(string[] args)
        {
            var context = new MyDbContext("MyDb");
            foreach(Student s in CreateStudents())
            {
                context.Students.Add(s);
            }
            context.SaveChanges();

            var students = context.Students.Where(s => s.GCSEGrade > 7).OrderBy(s => s.FullName);

            foreach (var student in students)
            {
               Console.WriteLine(student.Summary());
            }
            Console.ReadKey();
        }

        static List<Student> CreateStudents() {
            var list = new List<Student>();
            list.Add(new Student("Olivia", 345, Sex.Female, 8));
            list.Add(new Student("Noah", 744,  Sex.Male, 6));
            list.Add(new Student("Emma", 219,  Sex.Female, 7));
            list.Add(new Student("Adi", 700,  Sex.Male, 9));
            list.Add(new Student("Charlotte", 345,  Sex.Female, 6));
            list.Add(new Student("Sebastian", 744,  Sex.Male, 5));
            list.Add(new Student("James", 219,  Sex.Male, 8));
            list.Add(new Student("Mia", 700,  Sex.Female, 9));
            return list;
        }

    }

  
}

File: IntroToLINQ/IntroToLINQ_VB/Program - with EF.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace IntroToLINQ
    Class Program
        Private Shared Sub Main(ByVal args As String())
            Dim context = New MyDbContext("MyDb")

            For Each s As Student In CreateStudents()
                context.Students.Add(s)
            Next

            context.SaveChanges()
            Dim students = context.Students.Where(Function(s) s.GCSEGrade > 7).OrderBy(Function(s) s.FullName)

            For Each student In students
                Console.WriteLine(student.Summary())
            Next

            Console.ReadKey()
        End Sub

        Private Shared Function CreateStudents() As List(Of Student)
            Dim list = New List(Of Student)()
            list.Add(New Student("Olivia", 345, Sex.Female, 8))
            list.Add(New Student("Noah", 744, Sex.Male, 6))
            list.Add(New Student("Emma", 219, Sex.Female, 7))
            list.Add(New Student("Adi", 700, Sex.Male, 9))
            list.Add(New Student("Charlotte", 345, Sex.Female, 6))
            list.Add(New Student("Sebastian", 744, Sex.Male, 5))
            list.Add(New Student("James", 219, Sex.Male, 8))
            list.Add(New Student("Mia", 700, Sex.Female, 9))
            Return list
        End Function
    End Class
End Namespace
